extension/encoding/awslogsencodingextension/internal/unmarshaler/vpc-flow-log/unmarshaler.go

```diff
@@ -78,91 +103,94 @@ func (v *vpcFlowLogUnmarshaler) UnmarshalAWSLogs(reader io.Reader) (plog.Logs, e
 	}
 }
 
-// resourceKey stores the account id and region
-// of the flow logs. All log lines inside the
-// same S3 file come from the same account and
-// region.
-//
-// See https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs-s3-path.html.
-type resourceKey struct {
-	accountID string
-	region    string
-}
-
 func (v *vpcFlowLogUnmarshaler) unmarshalPlainTextLogs(reader io.Reader) (plog.Logs, error) {
-	scanner := bufio.NewScanner(reader)
+	// use buffered reader for efficiency and to avoid any size restrictions
+	bufReader := bufio.NewReader(reader)
```

Comment:
Did this cause any change on the benchmark performance? Can you check please? Switching from the `scanner` to the `bufio.Reader`